### PR TITLE
Improvements in Localhost environment

### PR DESCRIPTION
**Pull request by Wiezzel**
* `LocalhostPrerequisites` contain functions instead of return values because those were not flexible enough for more complicated test scenarios (e.g. timeouts, errors).
* Running GRPC server in a subprocess instead of a thread provides a great speed up and allows to implement proper `stop()`.

### CODE REVIEW

**Comment by Wiezzel** [comment]
Actually, `LocalhostRuntime.stop` still isn't working but let's ignore this for now.

File: tests/golem/envs/localhost.py [hunk 1]
```diff
@@ -168,17 +168,17 @@ def _spawn_server(self) -> None:
...
         self._started()
         return defer.succeed(None)
 
     def stop(self) -> defer.Deferred:
-        assert self._server is not None
         assert self._server_loop is not None
         assert self._server_thread is not None
         try:
-            self._server_loop.run_until_complete(self._server.stop())
+            # FIXME: This raises 'Cannot close a running event loop'
```
Comment (maaktweluit) [inline tests/golem/envs/localhost.py]: did you try to await stop first?

Comment (Krigpl) [reply tests/golem/envs/localhost.py]: Alternatively, you can try not calling `close` altogether since afaiu it's not necessary and `stop` should suffice.

Comment (Wiezzel) [reply tests/golem/envs/localhost.py]: @maaktweluit How can I await stop? `call_soon_threadsafe` doesn't return an awaitable object.

Comment (maaktweluit) [reply tests/golem/envs/localhost.py]: ahh, that is a bummer.. did not check the docs, was just a random thought

all examples from a quick search run close in the finally of the try block. Maybe we can move stop and close to the same function and use `.call_soon_threadsafe(` only once?

Comment (Wiezzel) [reply tests/golem/envs/localhost.py]: >Maybe we can move stop and close to the same function and use .call_soon_threadsafe( only once?

Comment (maaktweluit) [reply tests/golem/envs/localhost.py]: hahahaha

Ok, i'm out of ideas for now..

Comment (Wiezzel) [reply tests/golem/envs/localhost.py]: Went with multiprocessing in the end so this is deprecated.

**Review by Krigpl — APPROVED** [review]
lgtm, please update the summary to reflect what's really changed

**Pull request merged**